core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -291,23 +291,23 @@ class DynamicBrokerConfig(private val kafkaConfig: KafkaConfig) extends Logging
     dynamicDefaultConfigs.clone()
   }
 
-  private[server] def updateBrokerConfig(brokerId: Int, persistentProps: Properties): Unit = CoreUtils.inWriteLock(lock) {
+  private[server] def updateBrokerConfig(brokerId: Int, persistentProps: Properties, doLog: Boolean = false): Unit = CoreUtils.inWriteLock(lock) {
```

Comment:
I think in order to not modify the behavior in `ConfigHandler`, we need to make the default `doLog` as true, and then explicitly set it in line 215 above as `false`.